Fix DietPi-Drive_Manager when 'cancel' is selected on the drive list

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Drive_Manager | Fix DietPi-Drive_Manager when 'cancel' is selected on the drive list

nonnorm committed:
v6.35
Fix DietPi-Drive_Manager when 'cancel' is selected on the drive list

MichaIng commented on dietpi/dietpi-backup:
```diff
@@ -478,7 +478,7 @@ _EOF_
 					rm /tmp/dietpi-drive_manager_selmnt
 
-					[[ $FP_TARGET == '/' ]] && FP_TARGET='/mnt'
+					$FP_TARGET || [[ $FP_TARGET == '/' ]] && FP_TARGET='/mnt'
```
This would execute what was ever entered as command. You want to replace it with `/mnt` if the input was empty as well? That would be:
```
[[ ! $FP_TARGET || $FP_TARGET == '/' ]] && FP_TARGET='/mnt'
```

But actually, I think it's reasonable to not apply any choice when user cancels it, instead return to the menu. Let me see ich `/boot/dietpi/dietpi-drive_manager 1` returns an ext code on cancel, so that we can check that directly.

MichaIng committed:
v6.35

+ DietPi-Backup | Do not apply and check the file system when "Cancel" was selected in any of the target location selection menus
+ DietPi-Backup | Coding: Instead of creating a variable to check back indirectly, make Check_Supported_Directory_Location return it's result via exit code and check it back directly.
+ DietPi-Backup | Coding: Instead of using a large indented if-then-else block in Backup and Restore functions for the Check_Supported_Directory_Location result, return to menu directly, when the file system is not supported. For readability, other check should be coded in the same way, but it's enough for today :).

MichaIng commented:
Lol, I couldn't hold myself and did a larger code restructure. I don't like these huge if-then-else blocks til the end of the function, when it can be simply returned at the top based on condition.

But the relevant part: When "Cancel" is selected in any of the location selection menus, it is neither good to revert to `/mnt`, nor to further check the empty value, but now it is returned directly back to the menu without any change, which is what one would expect when hitting "Cancel".

MichaIng committed:
v6.35

+ DietPi-Backup | When searching for previous backups as target location, skip the supported file system check and do not touch the target directory variable
+ DietPi-Backup | Pre-select "Help" main menu entry initially
+ DietPi-Backup | Since the exit menu verification requires interactive confirmation, show main menu only when interactivity is guaranteed. Else it is possible to run into an endless loop via "G_INTERACTIVE=0 dietpi-backup" or when calling the script without "1"/"-1" arguments from cron job, systemd unit or else without valid terminal at STDIN.

MichaIng committed:
v6.35

+ DietPi-Backup | Syntax

MichaIng committed:
v6.35

+ DietPi-Backup | Do not check for mounted dietpi_userdata during supported target location check. It cannot be used as target location anyway, regardless of mount state, and if it is a mount/symlink, it is currently excluded from backup anyway. If/when we change this behaviour, to (optionally) include external drives/userdata into the system backup, mount states in general should be checked, and in case unmounted drives need to be excluded for the backup/restore run to neither remove an intact backup of a currently unmounted drive, nor restore an external drive backup to the root file system.

MichaIng commented:
Okay, added some other small enhancements and hardenings while testing up and down navigating through the menus and testing various valid and invalid inputs. It's more solid now with less unintended and unnecessary behaviour 😀.
